src/Parsers/ASTColumnDeclaration.cpp

```diff
@@ -78,7 +79,12 @@ void ASTColumnDeclaration::formatImpl(const FormatSettings & settings, FormatSta
     if (default_expression)
     {
         settings.ostr << ' ' << (settings.hilite ? hilite_keyword : "") << default_specifier << (settings.hilite ? hilite_none : "");
-        if (default_specifier != "EPHEMERAL" || !default_expression->as<ASTLiteral>()->value.isNull())
+        if (default_specifier != "EPHEMERAL" ||
+            !(
+                    default_expression->as<ASTLiteral>()->value.isNull() ||
+                    (type && default_expression->as<ASTLiteral>()->value == DataTypeFactory::instance().get(type)->getDefault())
+            )
+        )
```

Comment:
Weird formatting

Reply:
Also maybe store different parts in a boolean variable, it's really hard to follow what's going on.

Reply:
I refactored it - the previous implementation was quite deficient...